From c2b7bb7fd2d22c412fe13167073b6425b74768e6 Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Thu, 9 Oct 2014 15:15:05 +0800
Subject: [PATCH 1807/1825] mv_phy: Fix the error of the compile fail

commit b406647e8e38 [a38x: add initial mach-armada38x directory contents] from
https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
introduces the following compile error:

  CC      arch/arm/mach-armada38x/../plat-armada/mv_drivers_lsp/mv_phy/phy_sysfs.o
arch/arm/mach-armada38x/../plat-armada/mv_drivers_lsp/mv_phy/phy_sysfs.c:173:1: fatal error: opening dependency file arch/arm/mach-armada38x/../plat-armada/mv_drivers_lsp/mv_phy/.phy_sysfs.o.d: No such file or directory
compilation terminated.
make[3]: *** [arch/arm/mach-armada38x/../plat-armada/mv_drivers_lsp/mv_phy/phy_sysfs.o] Error 1
make[2]: *** [arch/arm/mach-armada38x] Error 2

Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/mach-armada38x/Makefile                   |    2 +-
 .../arm/plat-armada/mv_drivers_lsp/mv_phy/Makefile |    5 +++++
 2 files changed, 6 insertions(+), 1 deletions(-)
 create mode 100644 arch/arm/plat-armada/mv_drivers_lsp/mv_phy/Makefile

diff --git a/arch/arm/mach-armada38x/Makefile b/arch/arm/mach-armada38x/Makefile
index b47d529..cbcb420 100644
--- a/arch/arm/mach-armada38x/Makefile
+++ b/arch/arm/mach-armada38x/Makefile
@@ -94,7 +94,7 @@ obj-$(CONFIG_MV_INCLUDE_SWITCH)		+= $(QD_OBJS)
 # drivers part
 
 
-obj-$(CONFIG_MV_INCLUDE_GIG_ETH)	+= $(LSP_PHY_DIR)/phy_sysfs.o
+obj-$(CONFIG_MV_INCLUDE_GIG_ETH)	+= $(LSP_PHY_DIR)/
 
 obj-$(CONFIG_MV_USE_XOR_ENGINE)		+= $(PLAT_DRIVERS)/mv_xor/
 obj-$(CONFIG_MV_CESA)			+= $(PLAT_DRIVERS)/mv_cesa/
diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_phy/Makefile b/arch/arm/plat-armada/mv_drivers_lsp/mv_phy/Makefile
new file mode 100644
index 0000000..604145e
--- /dev/null
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_phy/Makefile
@@ -0,0 +1,5 @@
+ifneq ($(MACHINE),)
+include $(srctree)/$(MACHINE)/config/mvRules.mk
+endif
+
+obj-$(CONFIG_MV_INCLUDE_GIG_ETH) 	+= phy_sysfs.o
-- 
1.7.5.4

